genesis/engine/entities/rigid_entity/rigid_link.py

```diff
@@ -121,83 +122,77 @@ def _build(self):
         for vgeom in self._vgeoms:
             vgeom._build()
 
-        self._init_mesh = self._compose_init_mesh()
-
-        # inertial_mass and inertia_i
-        if self._inertial_mass is None:
-            if len(self._geoms) == 0 and len(self._vgeoms) == 0:
-                self._inertial_mass = 0.0
+        if self._inertial_mass is None or self._inertial_pos is None or self._inertial_i is None:
+            # Determine which geom list to use: geoms first, then vgeoms, then fallback
+            if len(self._geoms) > 0:
+                geom_list = self._geoms
+                is_visual = False
+            elif len(self._vgeoms) > 0:
+                gs.logger.info(
+                    f"Link mass is not specified and collision geoms can not be found for "
+                    f"link '{self.name}'. Using visual geoms to compute inertial properties."
+                )
+                geom_list = self._vgeoms
+                is_visual = True
             else:
-                if self._init_mesh.is_watertight:
-                    self._inertial_mass = self._init_mesh.volume * self.entity.material.rho
-                else:  # TODO: handle non-watertight mesh
-                    self._inertial_mass = 1.0
+                gs.logger.info(
+                    f"Link mass is not specified and no geoms found for "
+                    f"link '{self.name}'. Inertial and mass are set to zero."
+                )
+                self._inertial_mass = gs.EPS
+                self._inertial_pos = np.zeros(3, dtype=gs.np_float)
+                self._inertial_i = np.eye(3, dtype=gs.np_float) * gs.EPS
+                geom_list = []
+
+            # Process each geom individually and compose their properties
+            if len(geom_list) > 0:
+                total_mass = gs.EPS
```

Comment:
Can you add a comment to explain what using EPS mass is necessary? It does not make sense physically so it must be strongly motivated.